### PR TITLE
test: improve utils coverage

### PR DESCRIPTION
**Pull request by thaisguigon**
- Add some tests to improve utils coverage
- Add JSDoc comments to document some utils
- Replace `isDefined` and `assertNotNull` utils with `isNonNullable`

### CODE REVIEW

File: packages/twenty-front/src/utils/isDefined.ts [hunk 1]
```diff
@@ -1,3 +1,2 @@
-export const isDefined = <T>(
-  value: T | undefined | null,
-): value is NonNullable<T> => value !== undefined && value !== null;
+export const isDefined = <T>(value: T): value is NonNullable<T> =>
+  typeof value !== 'undefined' && value !== null;
```
Comment (charlesBochet) [inline packages/twenty-front/src/utils/isDefined.ts:2]: value !== undefined ?
Also, I think we should make the name more explicit: isDefinedAndNotNull

File: packages/twenty-front/src/utils/parseApolloStoreFieldName.ts [hunk 2]
```diff
@@ -8,19 +8,20 @@ export const parseApolloStoreFieldName = <
...
 ) => {
-  const matches = storeFieldName.match(/([a-zA-Z][a-zA-Z0-9 ]*)\((.*)\)/);
+  const matches = storeFieldName.match(/([a-zA-Z][a-zA-Z0-9 ]*)(\((.*)\))?/);
 
-  if (!matches?.[1]) return {};
+  const fieldName = matches?.[1];
 
-  const [, , stringifiedVariables] = matches;
-  const fieldName = matches[1] as string;
+  if (!fieldName) return {};
```
Comment (charlesBochet) [inline packages/twenty-front/src/utils/parseApolloStoreFieldName.ts:15]:
```
if (!fieldName) { 
  return {}
}
```

File: packages/twenty-front/src/utils/isDefined.ts (hunk 1 above)
Comment (charlesBochet) [reply packages/twenty-front/src/utils/isDefined.ts:2]: + add test to catch the 'undefined' mistake

**Review by charlesBochet — COMMENTED** [review]
Nice one, I have added two small comments

File: packages/twenty-front/src/utils/isDefined.ts (hunk 1 above)
Comment (thaisguigon) [reply packages/twenty-front/src/utils/isDefined.ts:2]: I changed it to use `isUndefined` and `isNull` from `@sniptt/guards` instead!
Also, I renamed `isDefined` to `isNonNullable`, as the return value is `value is NonNullable<T>`, `NonNullable` being defined as follows:
```
/**
 * Exclude null and undefined from T
 */
type NonNullable<T> = T & {};
```
And I changed the tests a little, though the `undefined` case was already included.

File: packages/twenty-front/src/utils/parseApolloStoreFieldName.ts (hunk 2 above)
Comment (thaisguigon) [reply packages/twenty-front/src/utils/parseApolloStoreFieldName.ts]: Done

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**